includes/class-wc-stripe-intent-controller.php

```diff
@@ -558,7 +568,11 @@ public function update_order_status_ajax() {
 			}
 			$save_payment_method = isset( $_POST['payment_method_id'] ) && ! empty( wc_clean( wp_unslash( $_POST['payment_method_id'] ) ) );
 
-			$gateway = new WC_Stripe_UPE_Payment_Gateway();
+			$gateway = $this->get_gateway();
+			if ( ! is_a( $gateway, 'WC_Stripe_UPE_Payment_Gateway' ) ) {
+				throw new WC_Stripe_Exception( __( "We're not able to process this payment.", 'woocommerce-gateway-stripe' ) );
```

Comment:
Probably we should use this to keep things consistent:
```suggestion
				throw new Exception( __( "We're not able to process this payment.", 'woocommerce-gateway-stripe' ) );
```
alternatively we could update all the above to use a common new method like:
```php
protected function get_upe_gateway() {
	$gateway = $this->get_gateway();
	if ( ! $gateway instanceof WC_Stripe_UPE_Payment_Gateway ) ) {
		throw new Exception( __( "We're not able to process this payment.", 'woocommerce-gateway-stripe' ) );
	}
	return $gateway;
}
```

Also I'm a fan of using `instanceof` instead of `is_a()` but they are virtually identical.

Reply:
I like the `get_upe_gateway()` function, great idea 👍🏼